src/controllers/legacycontrollermapping.h

```diff
@@ -143,6 +204,55 @@ class LegacyControllerMapping {
         return m_deviceDirection;
     }
 
+#ifdef MIXXX_USE_QML
+    /// Adds a custom QML module file to the list of controller modules for this mapping.
+    /// @param dirinfo A FileInfo of the directory or QML module
+    /// @param builtin If this is true, the script won't be written to the XML
+    virtual void addLibraryDirectory(const QFileInfo& dirinfo,
+            bool builtin = false) {
+        m_modules.append(QMLModuleInfo(
+                dirinfo,
+                builtin));
+        setDirty(true);
+    }
+
+    const QList<QMLModuleInfo>& getLibraryDirectories() const {
+        return m_modules;
+    }
+
+    /// @brief Adds a screen info where QML will be rendered.
+    /// @param identifier The screen identifier
+    /// @param size the size of the screen
+    /// @param targetFps the maximum FPS to render
+    /// @param splashoff the rendering grace time given when the screen is requested to shutdown
+    /// @param pixelFormat the pixel encoding format
+    /// @param endian the pixel endian format
+    /// @param reversedColor whether or not the RGB is swapped BGR
+    /// @param rawData whether or not the screen is allowed to reserve bare data, not transformed
+    virtual void addScreenInfo(const QString& identifier,
+            const QSize& size,
+            uint targetFps = 30,
+            uint splashoff = 50,
+            QImage::Format pixelFormat = QImage::Format_RGB32,
```

Comment:
default arguments considered harmful -- every place they are used in mixxx has caused problems over the years.  Just make these all explicit.